archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterSyntheticPrivateAccessesTest.java

```diff
@@ -422,6 +424,40 @@ Supplier<Supplier<Data_of_imports_private_constructor_reference_from_lambda.Targ
         );
     }
 
+    /**
+     * This is a special case, because for =+ concatenation of an outer string from an inner class
+     * the compiler creates a synthetic `access$123` method that creates a new `StringBuilder()`.
+     * Before this we wrongly assumed that all that happens from such `access$123` methods
+     * are field accesses and method calls, but no constructor calls.
+     */
+    @Test
+    public void imports_synthetic_access_from_string_concatenation() {
+        @SuppressWarnings("unused")
+        class Outer {
+            private String outerPrivateString = "Hello";
+
+            class Inner {
+                private void stringConcat() {
+                    outerPrivateString += ", world!";
+                }
+            }
+        }
+
+        Set<JavaFieldAccess> fieldAccesses = new ClassFileImporter().importClasses(Outer.class, Outer.Inner.class)
+                .get(Outer.Inner.class)
+                .getFieldAccessesFromSelf();
+
+        assertThatAccesses(fieldAccesses).contain(
+                access()
+                        .fromOrigin(Outer.Inner.class, "stringConcat")
+                        .toTarget(Outer.class, "outerPrivateString")
+                        .withAccessType(GET)).contain(
```

Comment:
Just a cosmetic comment, but I'd find this easier to read if the two `.contain` calls were at the same level, e.g.:
```java
        assertThatAccesses(fieldAccesses)
                .contain(access()
                        .fromOrigin(Outer.Inner.class, "stringConcat")
                        .toTarget(Outer.class, "outerPrivateString")
                        .withAccessType(GET)
                )
                .contain(access()
                        .fromOrigin(Outer.Inner.class, "stringConcat")
                        .toTarget(Outer.class, "outerPrivateString")
                        .withAccessType(SET)
                );
```

Reply:
Yikes, dunno how that happened 🙈 (the formatting wasn't intentional, maybe fixing a merge conflict or something...)